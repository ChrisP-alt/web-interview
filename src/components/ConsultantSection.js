import React from 'react'
import PropTypes from 'prop-types'

import { ConsultantTypes } from '../constants/Appointments'

const ConsultantSection = ({ selectConsultantType }) => {
  const buttons = Object.keys(ConsultantTypes).map(type => {
    const text = ConsultantTypes[type]
    return (
      <button
        className="button"
        key={type}
        onClick={() => selectConsultantType(type)}
        type="button"
      >
        {text}
      </button>
    )
  })

  return <div>{buttons}</div>
}

ConsultantSection.propTypes = {
  selectConsultantType: PropTypes.func.isRequired,
}

export default ConsultantSection
