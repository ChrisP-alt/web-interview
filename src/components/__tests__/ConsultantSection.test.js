import React from 'react'
import { shallow } from 'enzyme'

import ConsultantSection from '../ConsultantSection'

describe('ConsultantSection component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ConsultantSection
        selectConsultantType={jest.fn()}
        selectedConsultantType="gp"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('calls selectConsultantType when a button is clicked', () => {
    const selectConsultantType = jest.fn()
    const wrapper = shallow(
      <ConsultantSection
        selectConsultantType={selectConsultantType}
        selectedConsultantType="gp"
      />
    )
    const physioButton = wrapper.findWhere(node => node.key() === 'physio')
    physioButton.simulate('click')
    expect(selectConsultantType).toHaveBeenCalledWith('physio')
  })
})
