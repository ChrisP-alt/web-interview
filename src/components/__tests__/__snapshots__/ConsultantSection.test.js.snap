// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsultantSection component renders correctly 1`] = `
<Section
  Icon={[Function]}
  headerText="Consultant Type"
>
  <button
    className="buttonSelected"
    key="gp"
    onClick={[Function]}
    type="button"
  >
    GP
  </button>
  <button
    className="button"
    key="therapist"
    onClick={[Function]}
    type="button"
  >
    Therapist
  </button>
  <button
    className="button"
    key="physio"
    onClick={[Function]}
    type="button"
  >
    Physio
  </button>
  <button
    className="button"
    key="specialist"
    onClick={[Function]}
    type="button"
  >
    Specialist
  </button>
</Section>
`;
